Detectors/ITSMFT/ITS/workflow/src/CalibratorSpec.cxx

```diff
@@ -0,0 +1,1083 @@
+    recreate = true;
+  }
+
+  // Extract threshold values and save to memory
+  this->extract_thresh_row(chipID, this->currentRow[chipID]);
+  // Write thresholds to output data structures
+  this->update_output(chipID, recreate);
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+template <class Mapping>
+void ITSCalibrator<Mapping>::update_LHC_period(ProcessingContext& pc)
+{
+
+  const std::string LHCPeriodStr = pc.services().get<RawDeviceService>().device()->fConfig->GetProperty<std::string>("LHCPeriod", "");
+  if (!(LHCPeriodStr.empty())) {
+    this->LHC_period = LHCPeriodStr;
+  } else {
+    const char* months[12] = {"JAN", "FEB", "MAR", "APR", "MAY", "JUN",
+                              "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"};
+    time_t now = std::time(nullptr);
+    auto ltm = std::gmtime(&now);
+    this->LHC_period = months[ltm->tm_mon];
+    LOG(warning) << "LHCPeriod is not available, using current month " << this->LHC_period;
+  }
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+template <class Mapping>
+void ITSCalibrator<Mapping>::update_env_id(ProcessingContext& pc)
+{
+
+  const std::string envN = pc.services().get<RawDeviceService>().device()->fConfig->GetProperty<std::string>("environment_id", "");
+  if (!(envN.empty())) {
+    this->EnvironmentID = envN;
+  }
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+template <class Mapping>
+void ITSCalibrator<Mapping>::update_run_id(ProcessingContext& pc)
+{
+
+  const auto dh = DataRefUtils::getHeader<o2::header::DataHeader*>(
+    pc.inputs().getFirstValid(true));
+  if (dh->runNumber != 0) {
+    this->run_number = dh->runNumber;
+  }
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Main running function
+// Get info from previous stf decoder workflow, then loop over readout frames
+//     (ROFs) to count hits and extract thresholds
+template <class Mapping>
+void ITSCalibrator<Mapping>::run(ProcessingContext& pc)
+{
+
+  int lay, sta, ssta, mod, chip, chipInMod, rofcount; // layer, stave, sub stave, module, chip
+
+  auto orig = Mapping::getOrigin();
+
+  // Save environment ID and run number if needed
+  if (this->EnvironmentID.empty()) {
+    this->update_env_id(pc);
+  }
+  if (this->run_number == -1) {
+    this->update_run_id(pc);
+  }
+  if (this->LHC_period.empty()) {
+    this->update_LHC_period(pc);
+  }
+
+  // Calibration vector
+  const auto calibs = pc.inputs().get<gsl::span<o2::itsmft::GBTCalibData>>("calib");
+  const auto digits = pc.inputs().get<gsl::span<o2::itsmft::Digit>>("digits");
+  const auto ROFs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("digitsROF");
+  const auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(
+                           pc.inputs().get("digitsROF").header)
+                           ->startTime;
```

Comment:
Use 
```
const auto tfcounter = DataRefUtils::getHeader<DataProcessingHeader*>(pc.inputs().getFirstValid(true))->startTime;
```
instead of this line. Though, I don't see where do you use this counter.

Reply:
Thanks, we have switched to this line and commented it out. We were previously using this for print statements while debugging, so we will keep it (but commented out) for convenience while things are still in development.